synthetics/data_source_agents.go

```diff
@@ -43,7 +63,22 @@ func dataSourceAgentsRead(ctx context.Context, d *schema.ResourceData, m interfa
 	}
 
 	if resp.Agents != nil {
-		err = d.Set(itemsKey, agentsToMaps(*resp.Agents))
+		lat, latExists := d.GetOk(latitudeKey)
```

Comment:
Please extract `filterAgents(agents, d)` function to keep dataSourceAgentsRead() short.

Reply:
Addressed: b2aa92fd92caad336caa1da626aa32f7f5688243